packages/http-client-csharp/emitter/src/sdk-context-options.ts

```diff
@@ -8,19 +8,30 @@ export let defaultSDKContextOptions: CreateSdkContextOptions = {
 };
 
 /**
- * Sets the SDK context options to be used by the CSharp emitter.
- * @param options - The SDK context options.
+ * Sets the SDK context options to be used taking into account the passed in options and
+ * the default values used by the CSharp emitter.
+ * @param options - Emitter-specific options to be combined with the CSharp default options.
  * @beta
  */
 export function setSDKContextOptions(options: CreateSdkContextOptions) {
-  defaultSDKContextOptions = {
-    versioning: options.versioning ?? defaultSDKContextOptions.versioning,
+  defaultSDKContextOptions = getSDKContextOptions(options);
+}
+
+/**
+ * Gets the SDK context options to be used taking into account any passed in options and
+ * the default values used by the CSharp emitter.
+ * @param options - Optional emitter-specific options to be combined with the CSharp default options.
+ * @beta
+ */
+export function getSDKContextOptions(options?: CreateSdkContextOptions): CreateSdkContextOptions {
```

Comment:
Could you explain the usage of this function?
The user wants to get a combined option without changing the defaultSDKContextOptions?

Reply:
Yes, that was the point of it, but I have closed the PR as we don't actually need this method. We can just pass the options from the autorest emitter directly to the TCGC function without combining with the http-client-csharp defaults.

Reply:
And should we remove `setSDKContextOptions`? If we want the sub-emitter to call OnEmitter on its own, it should not affect MGC emitter.

Reply:
No, that is still necessary for the Azure plugin as it calls into MGCs onEmit function.